libs/@local/harpc/crates/wire/src/request/authorization.rs

```diff
@@ -0,0 +1,67 @@
+use std::io;
+
+use error_stack::Result;
+use graph_types::account::AccountId;
+use tokio::io::{AsyncRead, AsyncWrite};
+
+use crate::codec::{DecodePure, Encode};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(test, derive(test_strategy::Arbitrary))]
+pub struct Authorization {
```

Comment:
We could move this struct to a more sensible place (another crate) and use it more consistently. Nothing we require to do in this PR, though.